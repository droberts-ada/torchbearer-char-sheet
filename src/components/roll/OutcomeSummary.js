import React from 'react';
import PropTypes from 'prop-types';

import './styles/OutcomeSummary.css';

const OutcomeSummary = ({ outcome, ob, totalSuccesses }) => {
  return (
    <div className="outcome-summary">
      <h2>Outcome: {outcome}</h2>

      <div className="outcome-stats">
        <label htmlFor="ob">
          Obstacle
        </label>
        <label htmlFor="successes">
          Successes
        </label>
        <label htmlFor="margin">
          Margin
        </label>
        <span className="number" name="ob">
          {ob}
        </span>
        <span className="number" name="successes">
          {totalSuccesses}
        </span>
        <span className="number" name="margin">
          {totalSuccesses - ob}
        </span>
      </div>
    </div>
  );
};

OutcomeSummary.propTypes = {
  outcome: PropTypes.string.isRequired,
  ob: PropTypes.number.isRequired,
  totalSuccesses: PropTypes.number.isRequired,
}

export default OutcomeSummary;