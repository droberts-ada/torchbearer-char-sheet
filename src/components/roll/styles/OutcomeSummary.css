.outcome-summary h2 {
  font-size: 1.75em;
  text-transform: uppercase;
}

.outcome-summary .outcome-stats {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  justify-items: center;
}