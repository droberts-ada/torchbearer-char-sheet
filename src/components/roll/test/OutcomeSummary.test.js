import React from 'react';
import renderer from 'react-test-renderer';

import OutcomeSummary from '../OutcomeSummary';

describe('OutcomeSummary', () => {
  it('renders with a typical roll outcome', () => {
    const tree = renderer
      .create(
        <OutcomeSummary
          outcome="pass"
          ob={4}
          totalSuccesses={5}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})
