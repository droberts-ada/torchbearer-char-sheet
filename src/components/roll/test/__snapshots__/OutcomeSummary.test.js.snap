// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutcomeSummary renders with a typical roll outcome 1`] = `
<div
  className="outcome-summary"
>
  <h2>
    Outcome: 
    pass
  </h2>
  <div
    className="outcome-stats"
  >
    <label
      htmlFor="ob"
    >
      Obstacle
    </label>
    <label
      htmlFor="successes"
    >
      Successes
    </label>
    <label
      htmlFor="margin"
    >
      Margin
    </label>
    <span
      className="number"
      name="ob"
    >
      4
    </span>
    <span
      className="number"
      name="successes"
    >
      5
    </span>
    <span
      className="number"
      name="margin"
    >
      1
    </span>
  </div>
</div>
`;
